sandbox/apparmor/notify/listener/listener.go

```diff
@@ -326,8 +295,6 @@ func (l *Listener) Run() error {
 				break
 			}
 		}
-		// Ensure the tomb is killed before closing reqs channel
-		l.tomb.Kill(err)
```

Comment:
this might be a good idea anyway?

Reply:
Since `Run()` and `<-Reqs()` can't be on the same thread, and `Run()` doesn't return until all tracked goroutines return (including the main run loop), there's no reason to kill the tomb first, since the return value of `Run()` is now the only way the internal error is exposed, so no matter what the `reqs` channel will be closed first.

The `return err` two lines down kills the tomb anyway if `err != nil`, so I thought keeping the explicit `tomb.Kill()` could be misleading by making it seem important that that is the case, when it is not. Happy to revert this change though if you disagree.

Reply:
Actually, `reqs` could be closed by the outer `Run()` function rather than the inner goroutine. That doesn't actually change anything, since `Run()` still doesn't return until after, but if in the future we expose the error again, it can be more certain that the tomb is dying/dead with the given error before the `reqs` channel closes.